File: src/main/java/me/danetnaverno/editoni/io/MCAExtraInfo114.java
package me.danetnaverno.editoni.io;


import net.querz.nbt.tag.CompoundTag;

public class MCAExtraInfo114 extends MCAExtraInfo
{
    public MCAExtraInfo114(CompoundTag data)
    {
        super(data);
        CompoundTag level = data.getCompoundTag("Level");
        level.putByte("isLightOn", (byte) 0);
        level.remove("Sections");
        level.remove("Heightmaps");
        level.remove("Entities");
    }
}

File: src/main/kotlin/me/danetnaverno/editoni/io/MCAExtraInfo114.kt
package me.danetnaverno.editoni.io

import net.querz.nbt.tag.CompoundTag

class MCAExtraInfo114(data: CompoundTag) : MCAExtraInfo(data)
{
    init
    {
        val level = data.getCompoundTag("Level")
        level.putByte("isLightOn", 0.toByte()) //Invalidating light - Minecraft will recalculate it again
                                               //todo Neighboring chunks need light to be invalidated as well
        //level.remove("Sections")
        //level.remove("Heightmaps")
        //level.remove("Entities")
    }
}